exp/controllers/rosaroleconfig_controller.go

```diff
@@ -0,0 +1,662 @@
+	if err != nil {
+		conditions.MarkFalse(scope.RosaRoleConfig, expinfrav1.RosaRoleConfigReadyCondition, expinfrav1.RosaRoleConfigDeletionFailedReason, clusterv1.ConditionSeverityError, "Failed to delete account roles: %v", err)
+		return err
+	}
+
+	if scope.RosaRoleConfig.Spec.OperatorRoleConfig.OIDCID == "" {
+		err = r.deleteOIDCConfig(ocmClient, oidcID)
+		if err != nil {
+			conditions.MarkFalse(scope.RosaRoleConfig, expinfrav1.RosaRoleConfigReadyCondition, expinfrav1.RosaRoleConfigDeletionFailedReason, clusterv1.ConditionSeverityError, "Failed to delete OIDC config: %v", err)
+			return err
+		}
+	}
+
+	controllerutil.RemoveFinalizer(scope.RosaRoleConfig, expinfrav1.RosaRoleConfigFinalizer)
+	// Explicitly patch the object to persist the finalizer removal
+	if err := scope.PatchObject(); err != nil {
+		conditions.MarkFalse(scope.RosaRoleConfig, expinfrav1.RosaRoleConfigReadyCondition, expinfrav1.RosaRoleConfigDeletionFailedReason, clusterv1.ConditionSeverityError, "Failed to remove finalizer: %v", err)
+		return fmt.Errorf("failed to remove finalizer from ROSARoleConfig: %w", err)
+	}
+
+	return nil
+}
+
+func (r *ROSARoleConfigReconciler) createOperatorRoles(ctx context.Context, roleConfig *expinfrav1.ROSARoleConfig, scope *scope.RosaRoleConfigScope, ocmClient *ocm.Client) error {
+	installerRoleArn := scope.RosaRoleConfig.Status.AccountRolesRef.InstallerRoleARN
+	if installerRoleArn == "" {
+		return fmt.Errorf("installer role is empty")
+	}
+	oidcConfigID := scope.RosaRoleConfig.Status.OIDCID
+	if oidcConfigID == "" {
+		return fmt.Errorf("OIDCID is empty")
+	}
+
+	runtime := rosacli.NewRuntime()
+	policies, err := ocmClient.GetPolicies("OperatorRole")
+	if err != nil {
+		return err
+	}
+	runtime.OCMClient, err = rosa.NewOCMClient(ctx, scope)
+	if err != nil {
+		return err
+	}
+
+	runtime.Reporter = (&rosa.Reporter{})
+	runtime.Logger = rosalogging.NewLogger()
+	runtime.AWSClient, err = aws.NewClient().Logger(runtime.Logger).Build()
+	if err != nil {
+		return fmt.Errorf("failed to create aws client: %w", err)
+	}
+
+	runtime.Creator, err = runtime.AWSClient.GetCreator()
+	if err != nil {
+		return err
+	}
+
+	config := roleConfig.Spec.OperatorRoleConfig
+	version := roleConfig.Spec.AccountRoleConfig.Version
+	hostedCp := true
+	forcePolicyCreation := true
+	isSharedVpc := config.SharedVPCConfig.VPCEndpointRoleARN != "" && config.SharedVPCConfig.RouteRoleARN != ""
+
+	operatorRoles, err := runtime.AWSClient.ListOperatorRoles(version, "", config.Prefix)
+
+	if err != nil {
+		return err
+	}
+
+	if len(operatorRoles) > 0 {
+		for _, roles := range operatorRoles {
+			for _, role := range roles {
+				if strings.Contains(role.RoleName, fmt.Sprintf("%s-openshift-ingress-operator-cloud-credentials", config.Prefix)) {
+					scope.RosaRoleConfig.Status.OperatorRolesRef.IngressARN = role.RoleARN
+				}
+				if strings.Contains(role.RoleName, fmt.Sprintf("%s-openshift-image-registry-installer-cloud-credentials", config.Prefix)) {
```

Comment:
```suggestion
				 else if strings.Contains(role.RoleName, fmt.Sprintf("%s-openshift-image-registry-installer-cloud-credentials", config.Prefix)) {
```
better to set else if with the following if(s) check

Reply:
better to have the RoleNames in list/map so easier for you to check the roles count and existence